api/src/main/scala/org/virtuslab/ideprobe/protocol/Project.scala

```diff
@@ -1,9 +1,52 @@
 package org.virtuslab.ideprobe.protocol
 
+import java.nio.file.Path
+
+import org.virtuslab.ideprobe.ConfigFormat
+import org.virtuslab.ideprobe.jsonrpc.PayloadJsonFormat._
+import org.virtuslab.ideprobe.protocol.ContentRoot.{MainResources, MainSources, TestResources, TestSources}
+import pureconfig.error.CannotConvert
+import pureconfig.generic.auto._
+import pureconfig.{ConfigReader, ConfigWriter}
+
 case class Project(
     name: String,
     basePath: String,
     modules: Seq[Module]
 )
 
-case class Module(name: String, contentRoots: Seq[String], kind: Option[String])
+case class Module(name: String,
+                  contentRoots: Map[ContentRoot, Set[Path]],
+                  dependencies: Set[ModuleRef],
+                  kind: Option[String])
+
+object Module extends ConfigFormat {
+  implicit val reader: ConfigReader[Module] = exportReader[Module].instance
+  implicit val writer: ConfigWriter[Module] = exportWriter[Module].instance
+
+  import pureconfig._
+  import pureconfig.configurable._
+
+  implicit val contentRootsReader: ConfigReader[Map[ContentRoot, Set[Path]]] =
```

Comment:
wouldn't it be generated automatically (in some a bit less readable form, but still?

Reply:
Unfortunately it fails because it cannot convert `ContentRoot` to string and vice versa (even with manually specifying readers and writers for `ContentRoot`). It probably does not know it is being converted to string and not an arbitrary object.

Reply:
Oh, that's likely a limitation as in hocon map keys must be strings, and general mechanism of derivation would create objects for these (like `{ type = "MainSources" }`, so it couldn't be used as a key...